packages/components/src/autocomplete/types.ts

```diff
@@ -0,0 +1,205 @@
+/**
+ * WordPress dependencies
+ */
+import type { WPElement } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import type { useAutocomplete } from '.';
+
+type GetOptions< TCompleterOption = any > = (
+	filterValue: AutocompleterUIProps[ 'filterValue' ]
+) => Array< TCompleterOption > | Promise< Array< TCompleterOption > >;
```

Comment:
Something I've been meaning to ask before but I forgot about — could you point to where Autocomplete is receiving the `options` prop as a Promise?

Reply:
I can't actually, but looking back I'm pretty confident this typing came from[ the original README description](https://github.com/WordPress/gutenberg/blob/086b77ed409a70a6c6a6e74dee704851eff812f2/packages/components/src/autocomplete/README.md#L85-L86).

I need to think a bit on how this would impact the [logic](https://github.com/WordPress/gutenberg/blob/86a102f71383fb9d726fdccef7560064f3fc72b6/packages/components/src/autocomplete/get-default-use-items.tsx#L69) in `getDefaultUseItems` if a Promise promise were passed in.

Reply:
Calling `Promise.resolve()` on a promise [should be fine](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve#resolving_another_promise),

This also seems to match with definitely typed types:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__components/autocomplete/index.d.ts#L40

(the definitely typed way almost reads better, BTW — what do you think?)

Reply:
TIL about `PromiseLike`, thank you. That _does_ look a lot cleaner. e29c615f59